Prepare to release v0.9.0

djs55 opened the pull request:
- preliminary support for `Cstruct.t`
- add `topkg`
- modernize travis configuration

hannesm commented on tar-unix.opam:
```diff
@@ -8,20 +8,18 @@ dev-repo:     "https://github.com/mirage/ocaml-tar.git"
 
 depends: [
-  "jbuilder"          {build}
+  "jbuilder"          {build & >="1.0+beta7"}
```
I likely have a stupid question, but do the other opam files not require a `jbuilder >= "1.0+beta7"`?

djs55 replied:
I don't think that's a stupid question :) I resync'ed the `jbuild` files from `opam-repository` and only `tar-unix` seems to have that. I bet they should all have the `>= 1.0+beta7` really-- I'll adjust that.

hannesm commented on tar.opam:
```diff
@@ -8,11 +8,8 @@ dev-repo:     "https://github.com/mirage/ocaml-tar.git"
-  [ "jbuilder" "build" "--only-packages=tar" ]
-]
-
-build-test: [
```
this now means that the tests are never executed on travis!?

hannesm replied:
ups, sorry -- tests seem to be executed in the `tar-mirage` opam -- do the tests depend on tar-mirage?

djs55 replied:
The `lib_test/parse_test.ml` uses `Tar_mirage` (not ideal, probably the tests should be split up)

hannesm commented:
apart from minor nitpicks, I think this is great!  thanks for working on this!

hannesm commented on lib_test/jbuild:
```diff
@@ -1,6 +1,6 @@
 (executables
  ((names (parse_test))
-  (libraries (mirage-block-unix mirage-types-lwt oUnit lwt io-page.unix
+  (libraries (mirage-block-unix mirage-types-lwt oUnit lwt
```
I guess travis wants to tell you that `io-page-unix` is still needed somewhere in here...

djs55 commented:
One build is failing because `mirage-block-unix.2.8.3` doesn't build with `-safe-string`; fixed in https://github.com/mirage/mirage-block-unix/pull/83

I've added the `-safe-string` flag to the `jbuild` files here to be safe.

djs55 commented:
Since the only remaining problem is actually a failure of `mirage-block-unix` (fixed elsewhere) I think it's safe to do the release now.
